lib/evaluate/common.h

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FORTRAN_EVALUATE_COMMON_H_
+#define FORTRAN_EVALUATE_COMMON_H_
+
+#include "../semantics/enum-set.h"
```

Comment:
enum-set.h should probably move to a new library like "basic". idioms.h could go there too.
evaluate should need to depend on semantics.

Reply:
"should need to depend on semantics" should be "should *not* need to depend on semantics" above, yes?

I'll collect likely candidates for a common utility package and sweep them together.  Headers that don't depend on anything else in f18 will go there.

Reply:
Yes, should not.

That sounds like a good plan.